test_requirements.txt

```
@@ -44,7 +44,8 @@ psutil==5.8.0
 pyarrow>=1.0, <7.0
```

We need to lift this version cap. `>=1.0` doesn't add much and `<7.0` is breaking, so let's leave it free:

```suggestion
pyarrow
```

---

The reason is that pyarrow 6.0 has a broken build requirements specification https://pypi.org/project/pyarrow/6.0.1/#files

```toml
# pyproject.toml
[build-system]
requires = [
    "cython >= 0.29",
    "numpy==1.16.6; python_version<'3.8'",
    "numpy==1.17.3; python_version=='3.8'",
    "numpy==1.19.4; python_version=='3.9'",
    "numpy==1.21.3; python_version>'3.9'",  # <-- this is what causes the breakage
    "setuptools < 58.5",  # ARROW-14584
    "setuptools_scm",
    "wheel"
]
```